### DISCUSSION TITLE
Information on how to use this

### DISCUSSION THREAD

**Discussion by warm-considering** [Open-Sn/opensn#304] [Ask for Help] [answered]
Hello, I was wondering if there is a more detailed set of tutorials or perhaps some sort of API reference that details how to use Opensn/all the options available? I have managed to pull together a simple case based on the Kobayashi "dog leg" analytical benchmark by taking sections from the different test inputs. However I am not getting the results I expect and it is difficult to debug without more information on what the code is doing/what some of the input actually means. Any assistance would be appreciated.

**Comment by ragusa** [comment, marked as answer]
hello, thank you for reaching out. The problem you are trying to solve is difficult to solve using discrete-ordinates without using an uncollided-flux treatment. We are not planning on releasing such a method for another year or so. If you have access to a decent computer, using a locally refined angular quadrature (our SLDFE quadrature) is a decent alternative for the time being but this will never be as good and effiient as the uncollided flux formalism. The angular quadrature object is under refactoring right now, so no tutorial for the SLDFE are available until the lua interface is finalized. The syntax currently is (subject to change) for 3 cones of refinement in a certain direction (here [1,0,0]):
```
pquad = aquad.CreateSLDFESQAngularQuadrature(2)
aquad.LocallyRefineSLDFESQ(pquad,{1,0,0},45.0*math.pi/180,false)
aquad.LocallyRefineSLDFESQ(pquad,{1,0,0},23.0*math.pi/180,false)
aquad.LocallyRefineSLDFESQ(pquad,{1,0,0},12.0*math.pi/180,false)
```
Also, did you check if your mesh has the proper material layout, by doing a VTK export and plotting the material ID?

Finally, note that there isno official release as of today. We are planning on a v1 release for the second half of 2024.

**Reply by warm-considering** [reply]
Thank you for the reply, this is very useful info; I will try out the LSDFE quadrature. I look forward to the v1 release - I think this could be a very useful tool!
